postgres/osmap.yaml

```diff
@@ -91,9 +91,70 @@ RedHat:
 {% endif %}
 
 Suse:
+  pkg_repo:
+    name: pgdg-sles-{{ release }}
+    humanname: PostgreSQL {{ repo.version }} $releasever - $basearch
+    #Suse repos are suffixed 'suse/sles-12-$basearch' by upstream community convention.
+    baseurl: 'https://download.postgresql.org/pub/repos/yum/{{ repo.version }}/suse/sles-12-$basearch'
+    gpgkey:  'https://download.postgresql.org/pub/repos/yum/{{ repo.version }}/suse/sles-12-$basearch/repodata/repomd.xml'
+    #future: fix suse "4-Signatures public key is not available" issue and enable gpgcheck.
+    gpgcheck: 0
+
+{% if repo.use_upstream_repo %}
+
+  {% set lib_dir = '/var/lib/pgsql/' ~ repo.version ~ '/data' %}
+
+  pkg: postgresql{{ release }}-server
+  pkg_client: postgresql{{ release }}-server
+  conf_dir: {{ lib_dir }}
+  service: postgresql-{{ repo.version }}
+
+  prepare_cluster:
+    command: initdb --pgdata='{{ lib_dir }}'
+    test: test -f '{{ lib_dir }}/PG_VERSION'
+
+  # Directory containing PostgreSQL client executables
+  bin_dir: /usr/pgsql-{{ repo.version }}/bin
```

Comment:
Why to blindly copy-paste all the stuff? It seems that SUSE packages install all (?) needed alternatives itself during post-installation phase.

Reply:
The PR description explains why ... "Design is aligned with Redhat Stanza for convenience and continuity".  It keeps the PR simple.

>It seems that SUSE packages install all (?) needed alternatives itself during post-installation phase.
 
The upstream repo is common to SusE/RedHat/Fedora.  The alternatives handling can therefore be improved for these distributions but not in this PR.

Reply:
The SUSE section does need to be full copy of RedHat just for the sake of artificial consistency.
These options `bin_dir`, `client_bins` and `server_bins` serve a concrete purpose: allow subsequent states and `postges` modules to properly discover binaries and be able to do the job.

Other distros don't have them, because they don't need to. That's why it is configurable.
The necessary thing should be already done by installed packages, so the formula must not duplicate this.

Reply:
This alternatives cod is legacy. Perhaps in the past the packages did not already do this. 
 
I will investigate and raise common issue on RedHat/Suse.

Reply:
And Suse definitely doesn't require that.

Reply:
There has to be a better way to handle this than to cargo cult large blocks that are similar between os families

Reply:
Thanks for pointing me in right way here. Surely RedHat has same problem.  

Agreed - I tested and all except two are already handled by upstream packages.